tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/converter/PalladioTest.java

```diff
@@ -0,0 +1,99 @@
+import org.dataflowanalysis.analysis.pcm.PCMDataFlowConfidentialityAnalysisBuilder;
+import org.dataflowanalysis.analysis.pcm.core.seff.CallingSEFFActionSequenceElement;
+import org.dataflowanalysis.analysis.pcm.core.seff.SEFFActionSequenceElement;
+import org.dataflowanalysis.analysis.pcm.core.user.CallingUserActionSequenceElement;
+import org.dataflowanalysis.analysis.pcm.core.user.UserActionSequenceElement;
+import org.dataflowanalysis.analysis.testmodels.Activator;
+import org.dataflowanalysis.dfd.dataflowdiagram.DataFlowDiagram;
+import org.dataflowanalysis.dfd.dataflowdiagram.Node;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class PalladioTest {
+    @Test
+    @DisplayName("Test Ass to DFD")
+    public void assToDfd() {
+        String inputModel = "TravelPlanner";
+        String inputFile = "travelPlanner";
+        String modelLocation = "org.dataflowanalysis.analysis.testmodels";
+
+        final var usageModelPath = Paths.get("models", inputModel, inputFile + ".usagemodel").toString();
+        final var allocationPath = Paths.get("models", inputModel, inputFile + ".allocation").toString();
+        final var nodeCharPath = Paths.get("models", inputModel, inputFile + ".nodecharacteristics").toString();
+
+        DataFlowConfidentialityAnalysis analysis = new PCMDataFlowConfidentialityAnalysisBuilder().standalone().modelProjectName(modelLocation)
+                .usePluginActivator(Activator.class).useUsageModel(usageModelPath).useAllocationModel(allocationPath)
+                .useNodeCharacteristicsModel(nodeCharPath).build();
+
+        analysis.initializeAnalysis();
+        analysis.findAllSequences();
+        var sequences = analysis.findAllSequences();
+        var propagationResult = analysis.evaluateDataFlows(sequences);
+
+        Map<String, String> assIdToName = new HashMap<>();
+        for (ActionSequence as : propagationResult) {
+            for (AbstractActionSequenceElement<?> ase : as.getElements()) {
+                if (ase instanceof SEFFActionSequenceElement) {
+                    var cast = (SEFFActionSequenceElement<?>) ase;
+                    assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
+                } else if (ase instanceof CallingSEFFActionSequenceElement) {
+                    var cast = (CallingSEFFActionSequenceElement) ase;
+                    assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
+                } else if (ase instanceof CallingUserActionSequenceElement) {
+                    var cast = (CallingUserActionSequenceElement) ase;
+                    assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
+                } else {
+                    var cast = (UserActionSequenceElement<?>) ase;
+                    assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
+                }
```

Comment:
Use advanced instance of (e.g. putting the variable name after the type in instance of) makes this neater

Reply:
Would you mind giving me an example?

Reply:
You can write it like this:
```java
if (ase instanceof SEFFActionSequenceElement<?> cast) {
  assIdToName.putIfAbsent(cast.getElement().getId(), cast.getElement().getEntityName());
}
```

Reply:
Furthermore, there is another issue here:
- You can get access to the required attributes just by casting to `AbstractPCMActionSequenceElement` (notice the PCM)

Reply:
This is very neat